### ISSUE TITLE
Be able to deploy process application

### ISSUE THREAD

**Issue by philippfromme** [camunda/camunda-modeler#4667]
### Problem you would like to solve

Be able to deploy process applications.

### Proposed solution

* [ ] process application can be deployed (deploying all files, setting `bindingType` to `deployment`)
* [x] files can still be deployed individually
* [x] one deployment configuration per process application
* [x] to be decided
  * [x] is a _main process_ required
    * web modeler requires a _main process_
    * in web modeler only the _main process_ can be deployed
    * a main main process isn't required from the perspective of Zeebe, only from the perspective of the modeler to point to a process in Operate after deployment

# Deployment in Web Modeler

* in Web Modeler process applications can be deployed
* to deploy a process application the main process must be open
* all process application files are deployed automatically
* individual files cannot be deployed

![image](https://github.com/user-attachments/assets/576bdaf9-5f5a-4687-b62b-c3015b7a6b4a)

# Deployment in Desktop Modeler

* Camunda 7
  * no explicit process application concept
  * additional files can be added to deployment when deploying individual file
  * additional files must be added manually

![Image](https://github.com/user-attachments/assets/25dddac1-6f35-45ed-99c7-6141572fc64b)

* Camunda 8
  * only individual files can be deployed
  * camunda8-js-sdk [supports deploying multiple files](https://github.com/camunda/camunda-8-js-sdk/blob/e1d6eece8a18f8d8013c6db1a78463a5e12da778/src/zeebe/zb/ZeebeGrpcClient.ts#L797)

### Alternatives considered

None.

### Additional context

Related to https://github.com/camunda/camunda-modeler/issues/2508

**Comment by philippfromme** [comment]
I've been thinking about what the UI for this feature would look like.

![Image](https://github.com/user-attachments/assets/ff7e4f1e-f71d-4d34-9f74-f7a32d151c79)

Here's my idea:

* single file deployment should work the same as before
* if the file you're deploying is part of a process application offer a `Deploy process application` option that is set to `true` by default
* automatically list all of the files that will be deployed if `Deploy process application` is set to `true` (no need to add files manually like Camunda 7 deployment)
* deployment configuration could remain untouched for now and be saved per file or we could decide to save it per process application if `Deploy process application` is set to `true`

**Comment by nikku** [comment]
> if the file you're deploying is part of a process application offer a Deploy process application option that is set to true by default

Why offer such action? Process applications offer clear boundaries, and peace of mind. It does not make sense to not deploy certain parts of a process application, unless, i.e. they are explicitly excluded through a `.ignore` file.

**Comment by philippfromme** [comment]
Okay, I agree that it makes sense to align with Web Modeler.

**Comment by barmac** [comment]
> We could use the same approach.

I think it's a fair starting point. If there's need for include/exclude in the future, we can implement it. But first we need to be able to deploy the entire application.

**Comment by philippfromme** [comment]
I've created a couple of sketches.

Deploying a process application:

![Image](https://github.com/user-attachments/assets/1078f2de-1580-4a3e-a8ca-121e69024fe1)

Deploying a single file:

![Image](https://github.com/user-attachments/assets/e87d4047-ea06-42c8-8a9b-baaa02cf7f5b)

* added icons
* different titles
* number of resources shown when deploying process application

One unanswered question is whether we need the concept of a main process. When deploying a single file there's a notification with a link to the deployed file in Operate:

![Image](https://github.com/user-attachments/assets/dc0c8277-89f3-4d2c-8932-023653a5c9f5)

Without a main process I'm not sure what we'd link to. If we adopt the concept of a main process then the question would be whether you need to explicitly set it or whether it'll be the open file when deploying (which won't work if the open file isn't a BPMN file).

![Image](https://github.com/user-attachments/assets/3e4cff32-d217-42bb-a251-822c1015a5be)

Also, when there's a main process, will you be able to deploy if the open file isn't the main process?

![Image](https://github.com/user-attachments/assets/a5418830-63aa-431c-8743-04177264eb87)

Feedback welcome.

**Comment by nikku** [comment]
> One unanswered question is whether we need the concept of a main process. When deploying a single file there's a notification with a link to the deployed file in Operate:

I think we don't need it. The main process, if it exists, is an internal concept. Consider the following:

* You run a process from the modeler => Context decides what you run.
* You execute a decision from the modeler => Same thing
* You deploy a resource, and with it the rest of the process application => Link to the respective resource in operate.

![Image](https://github.com/user-attachments/assets/9862d091-212a-4229-919c-5b0ba3db2cd8)

No. Why would we want to do this? Let us not blindly replicate web modeler, where we can do better, let's do it.

**Comment by philippfromme** [comment]
Currently, we do not show any Operate link when deplyoying a DMN or form:

<img width="628" alt="Image" src="https://github.com/user-attachments/assets/a492d4c2-ad76-4846-a298-895fb70757c7" />

<img width="628" alt="Image" src="https://github.com/user-attachments/assets/b28821d7-5030-4fbf-8be8-95279985e171" />

We could keep the existing behavior and simply show the link to the currently open BPMN or no link at all.

**Comment by philippfromme** [comment]
> We could move it around once we rework the overall deploy experience

Yes, I think there is a lot to be improved but I want to keep the changes minimal for now. Lack of space is one of the main problems I think.

**Comment by lmbateman** [comment]
Deploying files individually: I agree that files inside a process application shouldn’t be individually deployable. We can validate this with users.

Main process:
- Nico’s argument in Slack against having a main process makes sense (although I’d like to see an example of a process application where the main process is not clear, because my brain kind of wants to have a main process).
- I agree that if a user deploys a resource, and the process application along with it, the link to Operate should open the resource that the user deployed.

Again, these are things we can validate with users.

Sketches:
- I like the headers with the icons; I think it makes it clear what you're deploying.
- We can probably shorten the text under the process application header. Ideally, I'd write, "14 resources will be deployed", but you're not supposed to start a sentence with numerals, and "Fourteen..." kind of defeats the purpose. What about, "Deploying 14 resources. Learn more."? Then, as Nico suggested, we could make the sentence expandable to show the list of resources, perhaps with interactions, such as an "x" to automatically add the file to an `excludes` directive.

**Comment by nikku** [comment]
Yes. This is the name of the deployment, it could be _my test deployment_. If Camunda 8 supports the name, let's be able to use it, also for multi-file deployments.

**Comment by philippfromme** [comment]
Yes, @remcowesterhoud confirmed that there is no such thing as a deployment name in C8. So I'm also in favor of just removing it.
